libs/src/api/account.js

```diff
@@ -100,91 +100,58 @@ class Account extends Base {
-          phase = phases.CREATE_USER_RECORD
-          await this.identityService.createUserRecord(email, this.web3Manager.getWalletAddress())
-        } else {
-          // Creator and identity service web3
-          this.REQUIRES(Services.CREATOR_NODE, Services.IDENTITY_SERVICE, Services.HEDGEHOG)
-
-          // If an owner wallet already exists, don't try to recreate it
-          if (!hasWallet) {
-            phase = phases.HEDGEHOG_SIGNUP
-            const ownerWallet = await this.hedgehog.signUp(email, password)
-            await this.web3Manager.setOwnerWallet(ownerWallet)
-            await this.generateRecoveryLink({ handle: metadata.handle, host })
-          }
-
-          phase = phases.UPLOAD_PROFILE_IMAGES
-          metadata = await this.User.uploadProfileImages(profilePictureFile, coverPhotoFile, metadata)
-
-          phase = phases.ADD_CREATOR
-          userId = await this.User.addCreator(metadata)
-        }
+      if (this.web3Manager.web3IsExternal()) {
+        this.REQUIRES(Services.CREATOR_NODE, Services.IDENTITY_SERVICE)
+
+        phase = phases.CREATE_USER_RECORD
+        await this.identityService.createUserRecord(email, this.web3Manager.getWalletAddress())
       } else {
-        if (this.web3Manager.web3IsExternal()) {
-          // Non-creator and external web3 (e.g. MetaMask)
-          this.REQUIRES(Services.IDENTITY_SERVICE)
-
-          phase = phases.UPLOAD_PROFILE_IMAGES
-          metadata = await this.User.uploadProfileImages(profilePictureFile, coverPhotoFile, metadata)
-
-          phase = phases.ADD_USER
-          userId = await this.User.addUser(metadata)
-
-          phase = phases.CREATE_USER_RECORD
-          await this.identityService.createUserRecord(email, this.web3Manager.getWalletAddress())
-        } else {
-          // Non-creator and identity service web3
-          this.REQUIRES(Services.IDENTITY_SERVICE, Services.HEDGEHOG)
-
-          // If an owner wallet already exists, don't try to recreate it
-          if (!hasWallet) {
-            phase = phases.HEDGEHOG_SIGNUP
-            const ownerWallet = await this.hedgehog.signUp(email, password)
-            await this.web3Manager.setOwnerWallet(ownerWallet)
-            await this.generateRecoveryLink({ handle: metadata.handle, host })
-          }
-
-          phase = phases.UPLOAD_PROFILE_IMAGES
-          metadata = await this.User.uploadProfileImages(profilePictureFile, coverPhotoFile, metadata)
-
-          phase = phases.ADD_USER
-          userId = await this.User.addUser(metadata)
+        this.REQUIRES(Services.CREATOR_NODE, Services.IDENTITY_SERVICE, Services.HEDGEHOG)
+
+        // If an owner wallet already exists, don't try to recreate it
+        if (!hasWallet) {
+          phase = phases.HEDGEHOG_SIGNUP
+          const ownerWallet = await this.hedgehog.signUp(email, password)
+          await this.web3Manager.setOwnerWallet(ownerWallet)
+          await this.generateRecoveryLink({ handle: metadata.handle, host })
```

Comment:
should recoveryLink be generated for both external/internal web3?

Reply:
we cant recover wallets for those who create users via not hedgehog